internal/controller/postgrescluster/pgbackrest.go

```diff
@@ -790,7 +790,7 @@ func (r *Reconciler) observeRestoreEnv(ctx context.Context,
 	restoreJobs := &batchv1.JobList{}
 	if err := r.Client.List(ctx, restoreJobs, &client.ListOptions{
 		LabelSelector: naming.PGBackRestRestoreJobSelector(cluster.GetName()),
-	}); err != nil {
+	}, client.InNamespace(cluster.GetNamespace())); err != nil {
```

Comment:
⛏️ The other hunks in this PR look like the following. Can we do the same here?

```diff
 	if err := r.Client.List(ctx, restoreJobs, &client.ListOptions{
+		Namespace: cluster.Namespace
 		LabelSelector: naming.PGBackRestRestoreJobSelector(cluster.GetName()),
 	}); err != nil {
```

Reply:
Yeah, so, as I said before, we are passing in list options in a lot of different ways throughout the codebase. I can go change this particular one if you want, but there will still be many other `Client.List` calls out there that don't have all of their list options neatly packaged in a `ListOptions` literal... Would you like for me to update all of them as well?

Reply:
No need to change everything. If it is currently using the struct, set the field. If it is passing the option arguments, pass `client.InNamespace()`. This stuck out because it is mixing the two styles.

Reply:
I hear ya. I was actually just copying what was done in two other `Client.List` calls later in the `observerRestoreEnv` method when I did it that way. I'll change all three.

Reply:
🤔 I see. What you have here matches ~L840 and ~L850. Not a blocker, but if you don't mind changing those to also use the struct "style", that's my preference.